apps/meteor/server/services/team/service.ts

```diff
@@ -472,8 +473,8 @@ export class TeamService extends ServiceClassInternal implements ITeamService {
 		room.teamDefault = isDefault;
 		await Rooms.setTeamDefaultById(rid, isDefault);
 
-		if (room.teamDefault) {
-			const teamMembers = await this.members(uid, room.teamId, true, undefined, undefined);
+		if (isDefault) {
+			const teamMembers = await this.members(uid, room.teamId, true, { offset: 0, count: MAX_NUMBER_OF_AUTO_JOIN_MEMBERS });
 
 			for await (const m of teamMembers.records) {
 				await addUserToRoom(room._id, m.user, user);
```

Comment:
```suggestion
				(await addUserToRoom(room._id, m.user, user)) && room.usersCount++;
```

I've just found out that this endpoint is returning an old (=incorrect) `usersCount` 
We return the room that has been fetched before, so it is not counting the users that have just been added. So that we don't need to search it again, we can increment the value as I suggested above
Please update the end-to-end tests to cover this